dbt/adapters/athena/impl.py

```diff
@@ -170,29 +171,41 @@ def is_work_group_output_location_enforced(self) -> bool:
         else:
             return False
 
-    @available
-    def s3_table_prefix(self, s3_data_dir: Optional[str]) -> str:
+    def _s3_table_prefix(self, s3_data_dir: Optional[str]) -> str:
         """
         Returns the root location for storing tables in S3.
-        This is `s3_data_dir`, if set, and `s3_staging_dir/tables/` if not.
+        This is `s3_data_dir`, if set at the model level, the s3_data_dir of the connection if provided,
+        and `s3_staging_dir/tables/` if nothing provided as data dir.
         We generate a value here even if `s3_data_dir` is not set,
         since creating a seed table requires a non-default location.
         """
         conn = self.connections.get_thread_connection()
         creds = conn.credentials
         if s3_data_dir is not None:
             return s3_data_dir
-        else:
-            return path.join(creds.s3_staging_dir, "tables")
+
+        if creds.s3_data_dir:
+            return creds.s3_data_dir
```

Comment:
why do we need this? I think that is redundant, _s3_table_prefix should be called already with s3_data_dir that is passed downstream from the different invocation isn't it?

Reply:
`generate_s3_location` is also called in `upload_s3_seeds` and we do not pass the `s3_data_dir` information

Reply:
Indeed I don't need it 👍🏻